src/libs/highlevel/elektra.c

```diff
@@ -0,0 +1,245 @@
+/**
+ * @file
+ *
+ * @brief Elektra High Level API.
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "elektra.h"
+#include <stdlib.h>
+#include <memory.h>
+#include "kdb.h"
+#include "kdbhelper.h"
+#include "elektraprivate.h"
+
+#define READ_KEY \
+Key * const nameKey = keyDup (elektra->parentKey); \
+keyAddName (nameKey, name); \
+\
+Key * const resultKey = ksLookup (elektra->config, nameKey, 0); \
+const char * string = keyString (resultKey); \
+\
+keyDel (nameKey); \
+
+#define RELEASE_KEY_AND_RETURN_VALUE \
+keyDel (resultKey); \
+\
+return value; \
+
+Elektra * elektraOpen (const char * application)
+{
+    Key * const parentKey = keyNew (application, KEY_END);
+    KDB * const kdb = kdbOpen (parentKey);
+
+    KeySet * const config = ksNew (0, KS_END);
+    kdbGet (kdb, config, parentKey);
+
+    Elektra * const elektra = elektraCalloc (sizeof (struct _Elektra));
+    elektra->kdb = kdb;
+    elektra->parentKey = parentKey;
+    elektra->config = config;
+
+    return elektra;
+}
+
+void elektraClose (Elektra * elektra)
+{
+    kdbClose (elektra->kdb, elektra->parentKey);
+    ksDel (elektra->config);
+    keyDel (elektra->parentKey);
+
+    elektraClearError (elektra);
+    elektraFree (elektra);
+}
+
+kdb_boolean_t elektraHasError (const Elektra * elektra)
+{
+    return elektra->error != NULL;
+}
+
+const char * elektraErrorMessage (const Elektra * elektra)
+{
+    return elektra->error->msg;
+}
+
+void elektraClearError (Elektra * elektra)
+{
+    elektraFree (elektra->error);
+    elektra->error = NULL;
+}
+
+/**
+ * @param elektra The elektra instance initialized with the parent key.
+ * @param name The keyname to look up. The keyname is appended to the parent key.
+ */
+const char * elektraGetString (Elektra * elektra, const char * name)
+{
+    READ_KEY
+
+    char * value = elektraMalloc (keyGetValueSize(resultKey));
+    strcpy (value, string);
+
+    RELEASE_KEY_AND_RETURN_VALUE
+}
+
+/**
+ * @param elektra The elektra instance initialized with the parent key.
+ * @param name The keyname to look up. The keyname is appended to the parent key.
+ */
+kdb_boolean_t elektraGetBoolean (Elektra * elektra, const char * name)
+{
+    READ_KEY
+
+    kdb_boolean_t value = 0;
+    if (!strcmp(string, "true") || !strcmp(string, "1") || !strcmp(string, "on")) {
```

Comment:
Here you introduce the same issue as discussed with hex values in #1388

We already have defined booleans to be `0` or `1`.

Furthermore, you need to check here if `check/type` isn't something else other than `boolean`.